### ISSUE TITLE
Replace votekick function by votespec

### ISSUE THREAD

**Issue by snailx3** [ddnet/ddnet#6866]
**Problem:** Some player groups are regularly kicking every other player off a server in order to accomplish a team 0 speedrun, giving Mods/Admin trouble moderating. 

**Our take:** We don't want to discourage team 0 speedrunning but we also don't want to offer the average player a bad experience by instantly getting kicked off a server.

**Previous discussions:** 
- Passwording a server can work but it feels like giving these players a privilege, creates confusion and completely relies on a staff member being available for a while. 
- Stoping the votekicks prevents the run from being perfected.
- Allowing random players to get kicked for blocking the team 0 speedrun is quite deceiving (sometimes they just don't speak the same language, making it impossible to explain the situation).

**Solution idea:** Get rid of the "votekick" feature at all and replace it with "votespec".
- It would work the same way the mod command "set_team" works: the tee gets killed and locked into spec for 1 minute. // **Contrarily to "set_team", the tee AND his dummy should both be sent to spec and not be able to revenge vote players while in spec**
- We also believe it will improve the overall experience for players as a votekick can almost always be solved by a votespec instead (even on full servers I'd arguewe shouldn't allow afk people to get kicked).

--------------

=> Any feedback/thoughts on the matter? 
Do you guys see any effective downside of getting rid of votekick?

**Comment by refeax** [comment]
The first question is, how long will these players stay in spec? 
I would suggest for 5 minutes, because votebans were also for 5 minutes, and it’s a reasonable time.

The second thing is, what if the player is spamming slurs or anything like that in the chat? I think this is where it would be smart to combine the set_team and the mute command, so that they will still be punished for inappropriate chat usage.

**Comment by snailx3** [comment]
About annoying/troll players, these guys usually just lock themselves in team to avoid being votekicked from normal players, so that would anyway require a staff member to intervene.

I get your point but I'm not sure if we should also mute them, it'd be hard for a normal player to emphasize that he understands what's going on / why he has been voted to spec if he can't answer back

Making the spec time 5 minutes sounds fine, I also think 1 minute is too short

**Comment by refeax** [comment]
Yea I was just thinking about those spam bots. They used to often get votebanned, and thus can’t spam anymore. But if they just get moved to spec, they can continue to spam. 
And now that i think of it, even with a 5 minute mute, if they’re not banned off the server, they’re just gonna continue to spam after these 5 minutes. That was my concern.

**Comment by l-ouis** [comment]
Maybe you could rephrase it instead of "Votespec" to "Vote Timeout" so the mute part makes more sense

**Comment by heinrich5991** [comment]
Another solution I'd see is implementing a team-0-like team. I.e. you can play in your own team, but don't get a team score. Normal team joining rules apply, but everyone gets individual times.

**Comment by heinrich5991** [comment]
> Annoying/troll players usually just lock themselves in team to avoid being votekicked from normal players which forces a staff member to intervene anyway.

That sounds like a separate problem we should fix, I guess. Maybe votekick could be downgraded to vote-mute for people in other teams?

**Comment by snailx3** [comment]
> > Annoying/troll players usually just lock themselves in team to avoid being votekicked from normal players which forces a staff member to intervene anyway.
> 
> That sounds like a separate problem we should fix, I guess. Maybe votekick could be downgraded to vote-mute for people in other teams?

Yes it is, I mostly mentioned it to highlight the fact that transforming votekick into votespec / votetimeout wouldn't deteriorate that existing problem - even if it doesn't address it. 

It makes sense that votekicking someone in team instead vote-mutes him, I'm for it too.

**Comment by snailx3** [comment]
> Another solution I'd see is implementing a team-0-like team. I.e. you can play in your own team, but don't get a team score. Normal team joining rules apply, but everyone gets individual times.

I thought about that one too, but I'm scared this will make big groups just play in their own team 0 and will completely split t0 maps into "pro groups but team 0-like which you can only join under invite" and "normal players struggling in standard team 0"
